src/Falanx.Machinery/Reflection.fs

```diff
@@ -121,10 +122,85 @@ module Reflection =
               get typ 
             else typ
             
+        let maxTuple = 8
+        
+        // Which field holds the nested tuple?
+        let tupleEncField = maxTuple - 1
+        
+        let simpleTupleNames = [|
+            "Tuple`1";      "Tuple`2";      "Tuple`3";
```

Comment:
maybe

```fsharp
for i in 1..8 do
   yield sprintf "Tuple`%i" i
   yield sprintf "ValueTuple`%i" i
```

Reply:
This is code ripped from the compiler so...

Reply:
Im going to leave as it is for now, the code generated form an array comprehension is about 5x more than the constant version.